homeassistant/components/yolink/entities.py

```diff
@@ -0,0 +1,110 @@
+"""Entity for YoLink devices."""
+from __future__ import annotations
+
+from homeassistant.components.binary_sensor import BinarySensorEntity
+from homeassistant.components.sensor import SensorEntity
+from homeassistant.const import TEMP_CELSIUS
+
+from .device import YoLinkDevice, YoLinkDeviceEntity
+
+SENSOR_TYPES_CONFIG = {
+    "DoorSensor": [None, None, "door"],
+    "LeakSensor": [None, None, "moisture"],
+    "MotionSensor": [None, None, "motion"],
+    "temperature": [TEMP_CELSIUS, None, "temperature"],
+    "humidity": [None, None, "humidity"],
+    "battery": [None, None, "battery"],
+}
+
+
+class YoLinkSensorEntity(YoLinkDeviceEntity, SensorEntity):
+    """Representation of a YoLink Sensor."""
+
+    def __init__(self, device: YoLinkDevice, entity_type, config_entry):
+        """Initialize the YoLink Sensor."""
+        YoLinkDeviceEntity.__init__(self, device, entity_type, config_entry)
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend."""
+        try:
+            return SENSOR_TYPES_CONFIG.get(self._type)[1]
+        except TypeError:
+            return None
+
+    @property
+    def native_unit_of_measurement(self):
+        """Return the unit of measurement of this entity, if any."""
+        try:
+            return SENSOR_TYPES_CONFIG.get(self._type)[0]
+        except TypeError:
+            return None
+
+    @property
+    def device_class(self):
```

Comment:
Please implement entity descriptions to describe the different entity types instead. See eg:

https://github.com/home-assistant/core/blob/fe3fb230bed57bb7ab130c0186db3ba59f792e60/homeassistant/components/airly/sensor.py#L59-L117